src/Illuminate/Support/Collection.php

```diff
@@ -59,6 +59,39 @@ public static function make($items = [])
     }
 
     /**
+     * Create a new collection from a multi-level array.
+     *
+     * @param  mixed  $items
+     * @return static
+     */
+    public static function makeDepth($items = [])
+    {
+        $collections = self::makeDepthRecursive($items);
+
+        return new static($collections);
+    }
+
+    /**
+     * Recursive iteration over arrayable mapping arrays to collections.
+     *
+     * @param mixed $item
+     * @return static
+     */
+    private static function makeDepthRecursive($items)
+    {
+        $items = collect()->getArrayableItems($items);
```

Comment:
I don't like this, but didn't know how to call `getArrayableItems` without having a working instance of a collection.